Wrong type checking in Aruco Board_create

The board_create python function implemented following #771 throws a type error

To reproduce the bug:

```.python
import cv2.aruco as aruco
import numpy as np

points = [np.array([(0,0,0),(1,0,0),(1,-1,0),(0,-1,0)],('float32'))]

arucoDict = aruco.Dictionary_get(aruco.DICT_4X4_50)

ids = np.array([[0]])

board = aruco.Board_create(points,arucoDict,ids)
```

will throw the following error:
```.bash
    _board = aruco.Board_create(_points,_arucoDict,_ids)
cv2.error: /home/forssea/opencv_contrib/modules/aruco/src/aruco.cpp:1281: error: (-215) objPoints.type() == CV_32FC3 in function create
```

Shouldn't the type to check be CV_32F instead of CV_32FC3?

--- a/modules/aruco/src/aruco.cpp
+++ b/modules/aruco/src/aruco.cpp
@@ -1278,7 +1278,7 @@ void GridBoard::draw(Size outSize, OutputArray _img, int marginSize, int borderB
 Ptr<Board> Board::create(InputArrayOfArrays objPoints, const Ptr<Dictionary> &dictionary, InputArray ids) {
 
     CV_Assert(objPoints.total() == ids.total());
-    CV_Assert(objPoints.type() == CV_32FC3);
+    CV_Assert(objPoints.type() == CV_32F);
 
     std::vector< std::vector< Point3f > > obj_points_vector;
     for (unsigned int i = 0; i < objPoints.total(); i++) {

--- a/modules/aruco/src/aruco.cpp
+++ b/modules/aruco/src/aruco.cpp
@@ -1278,12 +1278,20 @@ void GridBoard::draw(Size outSize, OutputArray _img, int marginSize, int borderB
 Ptr<Board> Board::create(InputArrayOfArrays objPoints, const Ptr<Dictionary> &dictionary, InputArray ids) {
 
     CV_Assert(objPoints.total() == ids.total());
-    CV_Assert(objPoints.type() == CV_32F);
+    CV_Assert(objPoints.type() == CV_32FC3 || objPoints.type() == CV_32FC1);
 
     std::vector< std::vector< Point3f > > obj_points_vector;
     for (unsigned int i = 0; i < objPoints.total(); i++) {
         std::vector<Point3f> corners;
         Mat corners_mat = objPoints.getMat(i);
+        if(objPoints.type() == CV_32FC1)
+        {
+            CV_Assert(corners_mat.total() == 12);
+        }
+        else
+        {
+            CV_Assert(corners_mat.total() == 4);
+        }
         for (int j = 0; j < 4; j++) {
             corners.push_back(corners_mat.at<Point3f>(j));
         }

